src/CloudFoundry.Manifests/Manifest.cs

```diff
@@ -0,0 +1,426 @@
+            }
+
+            return apps.ToArray();
+        }
+
+        /// <summary>
+        /// Save a manifest to disk
+        /// </summary>
+        /// <param name="contents">Array containing applications to be saved in the manifest</param>
+        /// <param name="path">Path to the manifest file</param>
+        public static void Save(Application[] contents, string path)
+        {
+            if (contents == null)
+            {
+                throw new ArgumentNullException("contents");
+            }
+            Dictionary<string, object> yaml = new Dictionary<string, object>();
+            List<Dictionary<string, object>> apps = new List<Dictionary<string, object>>();
+            foreach (Application app in contents)
+            {
+                Dictionary<string, object> appDict = new Dictionary<string, object>();
+                appDict.Add("name", app.Name);
+                appDict.Add("memory", string.Format(CultureInfo.InvariantCulture, "{0}M", app.Memory.ToString()));
+                appDict.Add("instances", app.InstanceCount);
+                if (!string.IsNullOrWhiteSpace(app.BuildpackUrl))
+                {
+                    appDict.Add("buildpack", app.BuildpackUrl);
+                }
+                if (app.HealthCheckTimeout != null && app.HealthCheckTimeout > 0)
+                {
+                    appDict.Add("timeout", app.HealthCheckTimeout);
+                }
+                if (!string.IsNullOrWhiteSpace(app.Command))
+                {
+                    appDict.Add("command", app.Command);
+                }
+                if (app.GetHosts() != null && app.GetHosts().Length > 0)
+                {
+                    appDict.Add("host", app.GetHosts()[0]);
+                    if (app.GetDomains() != null && app.GetDomains().Length > 0)
+                    {
+                        appDict.Add("domain", app.GetDomains()[0]);
+                    }
+                }
+                else
+                {
+                    appDict.Add("no-route", true);
+                }
+                if (app.GetServices() != null && app.GetServices().Length > 0)
+                {
+                    appDict.Add("services", app.GetServices().ToList<string>());
+                }
+                if (app.EnvironmentVariables != null && app.EnvironmentVariables.Count > 0)
+                {
+                    appDict.Add("env", app.EnvironmentVariables);
+                }
+                apps.Add(appDict);
+            }
+            yaml["applications"] = apps;
+            using (TextWriter writer = File.CreateText(path))
+            {
+                Serializer serializer = new Serializer();
+                serializer.Serialize(writer, yaml);
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Globalization", "CA1308:NormalizeStringsToUppercase")]
```

Comment:
Assuming this is because of this: if (match[0] == "${random-word}")
Are cf manifests case sensitive? If so, please add a justification to the suppress message